promoter/file/filestore.go

```diff
@@ -62,6 +61,25 @@ func openFilestore(
 	ctx context.Context,
 	filestore *api.Filestore,
 	useServiceAccount, confirm bool,
+) (syncFilestore, error) {
+	if strings.HasPrefix(filestore.Base, "gs://") {
+		return openGCSFilestore(ctx, filestore, useServiceAccount, confirm)
+	}
+	if strings.HasPrefix(filestore.Base, "s3://") {
+		return openS3Filestore(filestore)
+	}
+
+	return nil, fmt.Errorf(
+		"unrecognized scheme %q (supported schemes: s3, gs)",
+		filestore.Base,
+	)
```

Comment:
Let's define something like:

```go
var supportedURISchemes := []string{
  "s3",
  "gs",
}
```

And then:

```go
	return nil, fmt.Errorf(
		"unrecognized scheme %q (supported schemes: %s)",
		filestore.Base,
                strings.Join(supportedURISchemes, ", "),
	)
```

Reply:
Created a register of providers, so that everything stays in sync here.  I think that's the underlying concern, and I agree with it :+1: